apps/communications/contacts/js/bootstrap.js

```diff
@@ -0,0 +1,423 @@
+/* global applyCache */
+/* global LazyLoader */
+/* global utils */
+
+/* exports applyCache */
+
+'use strict';
+
+(function(exports) {
+
+  // Ideally we should store this value in the configuration file.
+  // However, that means that we need to load two more script files and
+  // read directly from the config.json file or from the cookie storing
+  // the configuration data. This steals us ~150ms of startup time, so
+  // we chose to hard code this value instead.
+  var CACHE_ENABLED = true;
+
+  var _observers = {};
+
+  function addObserver(aMessage, aCallback) {
+    if (!_observers[aMessage]) {
+      _observers[aMessage] = [];
+    }
+    _observers[aMessage].push(aCallback);
+  }
+
+  function notifyObservers(aMessage, aData) {
+    var observers = _observers[aMessage];
+    if (!observers) {
+      return;
+    }
+
+    for (var callback in observers) {
+      if (observers[callback] && typeof observers[callback] === 'function') {
```

Comment:
ni, we could do the `function` type checking on the `addObserver` function. So we are sure that we will have proper callbacks functions.